src/main/java/seedu/recruit/model/Model.java

```diff
@@ -178,6 +178,27 @@
      */
     void addJobOffer(CompanyName companyName, JobOffer jobOffer);
 
+    /**
+     * Returns true if a company has a job offer with the same identity as {@code jobOffer} exists in the CompanyBook.
+     */
+    boolean hasJobOffer(CompanyName companyName, JobOffer jobOffer);
+
+    /**
+     * Replaces the given job offer {@code target} in the list with {@code editedJobOffer}.
+     * {@code target} must exist in the company book.
+     * The job offer identity of {@code editedJobOffer} must not be the same as another existing job offer in the
+     * company book.
+     */
+    void updateJobOffer(JobOffer target, JobOffer editedJobOffer);
```

Comment:
probably can be better named to be more specific as it can be quite similar to updateJobOfferInCompany.
suggestion: updateJobOfferInCompanyBook vs updateJobOfferOfSelectedCompany

Reply:
Sure noted